examples/example_fastapi/main.py

```diff
@@ -163,53 +199,107 @@ async def ip_address():
 
 @app.get("/ip/4")
 async def connect_ipv4():
-    """Connect to the Quad9 VPN provider using their IPv4 address."""
-    ipv4_host = "9.9.9.9"
-    try:
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        sock.settimeout(5)
-        sock.connect((ipv4_host, 53))
-        return {"result": True}
-    except TimeoutError:
-        logger.warning(f"Socket connection for host {ipv4_host} failed")
-        return {"result": False}
+    """Connect to some DNS services using their IPv4 address.
+    The webserver on that address can return a 404 error, and it is normal, so we accept that response code.
+    """
+    ipv4_hosts: list[str] = [
+        "https://9.9.9.9",  # Quad9 VPN service
+        "https://1.1.1.1",  # CloudFlare DNS service
```

Comment:
Ali told me just this morning that 1.1.1.1 has issues on some French networks because they used 1.1.1.1 as the local router interface. Can we use Google's instead to avoid ever encountering this issue?